xform-anatomy.rst

```diff
@@ -0,0 +1,420 @@
+  <instance>
+   <data id="myform" >
+     <meta>
+       <instanceID/>
+     </meta>
+     <mystring/>
+   </data>
+  </instance>
+
+  <!-- bindings -->
+  <bind nodeset="/data/meta/instanceID" type="string" 
+           readonly="true()" calculate="concat('uuid:',uuid())" />
+  <bind nodeset="/data/mystring" type="string"/>
+
+  </model>
+  </h:head>
+
+  <!-- body -->
+  <h:body>
+  <input ref="/data/mystring">
+    <label>This is a string prompt</label>
+    <hint>You don't need a hint, but you can add one</hint>
+  </input>
+  </h:body>
+
+  </h:html>
+
+Instances
+----------
+
+.. code-block:: html
+
+  <instance>
+   <data id="myform" >
+    <meta>
+      <instanceID/>
+    </meta>
+    <mystring/>
+  </data>
+  </instance>
+
+Here we have a single instance called "data". This instance is identified via the 'id' attribute as the 'myform' form. This id attribute, and an optional version attribute, provide the internal unique name of this form to the software. The software only allows one form definition for each (id, version) pair.
+
+The form itself has two variables; the first, called "instanceID" is nested within a "meta" block. The second variable is called "mystring." Nesting within blocks can help organize your data; we could have nested "mystring" within one or more blocks. The "meta" block and "instanceID" variable are defined as part of a form design standard, and must appear as shown by OpenRosa 1.0 Metadata Schema.
+
+We are going to compute and store a unique identifier for each filled-in form in the ``<instanceID/>`` element. This will enable ODK Aggregate (or any back-end process) to de-duplicate submissions if, for example, the same filled-in form were somehow submitted twice.
+
+The other variable, ``<mystring/>``, is where we're going to store the answer to our prompt, but it's only a placeholder. We **DO NOT** specify a data type here.
+
+The first instance in the form represents a saved or completed version of the form. When you save a filled-in form, the XML output file will look something like this:
+
+.. code-block:: xml
+
+  <data id="myform" >
+    <meta>
+      <instanceID>uuid:23b56e39-ef50-4510-b85f-c454cd5465c1</instanceID>
+    </meta>
+    <mystring>Hello World!  This is my answer!</mystring>
+  </data>
+
+Beginning with ODK Collect 1.2, there can be multiple instances defined in the form. The second and subsequent instance definitions can specify static data used in filling out the form (e.g., lists of cities within each county within a given state, dosage tables). The most common use would be to present the choices in a cascading select, e.g., where you are asked to first choose a state, then the county within that state, then the city within that county. 
+
+If you specify a value for :``<mystring>`` in the original XForm, it presents that as the default answer to the prompt:
+
+.. code-block:: xml
+
+  <mystring>Default answer</mystring>
+
+Adding more prompts is as simple as adding more variables:
```

Comment:
These are not variables. They are fields (maybe) or nodes in the data structure. And "prompts" is also weird.

Maybe:
Each node in the Primary Instance represents a piece of data that will be collected by the form.